packages/babel-traverse/src/scope/index.ts

```diff
@@ -899,6 +870,21 @@ export default class Scope {
     };
 
     this.crawling = true;
+    // traverse does not visit the root node, here we explicitly collect
+    // root node binding info when the root is not a Program.
+    if (programParent.path !== path && collectorVisitor._exploded) {
+      // @ts-expect-error when collectorVisitor is exploded, `enter` always exists
+      for (const visit of collectorVisitor.enter) {
+        visit(path, state);
+      }
+      const typeVisitors = collectorVisitor[path.type];
+      if (typeVisitors) {
+        // @ts-expect-error when collectorVisitor is exploded, `enter` always exists
+        for (const visit of typeVisitors.enter) {
+          visit(path, state);
+        }
+      }
+    }
```

Comment:
In the future we might want to extract this as a generic utility to visit a single specific node.